cli/tests/test_git_remotes.rs

```diff
@@ -204,6 +225,15 @@ fn test_git_remote_rename() {
     bar http://example.com/repo/foo
     baz http://example.com/repo/baz
     "###);
+    let git_config = std::fs::read_to_string(repo_path.join(".jj/repo/store/git/config")).unwrap();
+    insta::assert_snapshot!(git_config[git_config.find("[remote ").unwrap()..], @r#"
+    [remote "baz"]
+    	url = http://example.com/repo/baz
+    	fetch = +refs/heads/*:refs/remotes/baz/*
+    [remote "bar"]
+    	url = http://example.com/repo/foo
+    	fetch = +refs/heads/*:refs/remotes/bar/*
+    "#);
 }
```

Comment:
Can you also add rename tests with `[branch ".."]` section and custom `fetch` refspecs?

`jj` doesn't use them, but colocated git repo may have configuration for `git` CLI. `libgit2` updates them.

https://github.com/libgit2/libgit2/blob/b042e574064859c01cb902a3f4f418a3706eb78d/src/libgit2/remote.c#L2558

Reply:
Sorry, I missed that this was about `branch.*` rather than `remote.*`. I’ll take a look at those. I think what I said still stands about weird refspecs in `remote.*`, though.

Reply:
> So the only change we could make here to improve compatibility with the previous implementation would be to preserve non‐default fetch refspecs and leave push refspecs untouched, which would make the ref renaming somewhat weird. Do you want me to do that, or just add a test to show that the behaviour has changed?

It's probably better to improve compatibility with `git` CLI, not with `libgit2`.

I know `libgit2` doesn't parse negative refspecs and errors out. It wouldn't silently remove unsupported refspecs, though.

Reply:
After thinking about it some more, I’ve changed the PR to detect when it would potentially clobber non‐standard configuration and error out instead; it shouldn’t silently throw away odd configuration any more. I think this is preferable to adding more logic to handle Git configuration that Jujutsu isn’t really compatible with anyway. People who have complex non‐standard Git remote configuration for use with a coloated repository should prefer to use `git remote` directly. Maybe it’d be good for us to support some of it in future (e.g. `pushurl`), but it seems out of scope for this PR. Let me know what you think and if you want me to test the failures cases here.

Reply:
> After thinking about it some more, I’ve changed the PR to detect when it would potentially clobber non‐standard configuration and error out instead; it shouldn’t silently throw away odd configuration any more. I think this is preferable to adding more logic to handle Git configuration that Jujutsu isn’t really compatible with anyway.

I agree that non-standard refspecs can be an error. However, upstream branches can be set by `git clone` and other GUI tooling. It would be annoying if `jj git remote` can't rename/delete remote because of the existence of `[branch ]`s.

Reply:
This now handles basic branch configuration. I think the state after this PR is still weird (bookmark renames break branch config, remote renames don’t), and I’m not a huge fan of the extra complexity, but it at least keeps us working with `git clone`‐generated configs in the same way as before the PR and rejects anything more elaborate than that. Let me know what you think.

We also don’t have tests to ensure that non‐standard configurations get rejected. I’m not sure if it’s worth it and I’m quite eager to finally get this PR landed, but I can add them if you’d like.